### DISCUSSION TITLE
embedded osgi.annotation.jar

### DISCUSSION THREAD

**Discussion by laeubi** [eclipse-equinox/equinox#130] [Q&A] [answered]
I noticed that equinox has osgi.annotation.jar embedded and used as jars.extra.classpath while other OSGi sources are just inside the bundle itself and compiled. Is there any reason for this? Binary files are hard to version and jars.extra.classpath is pde specific and therefore usually not accessible for other maven tools...

**Comment by tjwatson** [comment, marked as answer]
These annotations are compile time only annotations.  We should never introduce a solution that would require these packages to actually exist at runtime (i.e. a result that adds `Import-Package: org.osgi.annotation.versioning" to the bundles that compile against them)

But perhaps this is already fixed by PDE/tycho?  Looking at my PDE projects for Equinox it seems the "Plug-in Dependencies" classpath container already auto-adds:

plugins/org.osgi.annotation.bundle_2.0.0.202202082230.jar
plugins/org.osgi.annotation.versioning_1.1.2.202109301733.jar
plugins/org.osgi.service.component.annotations_1.5.0.202109301733.jar

I assume the same thing happens with a Tycho/maven build.  If that is the case then I believe we can safely remove our embedded osgi.annotation.jar and just assume they come from free?

**Reply by laeubi** [reply]
Yes the bundle annotations are now automatically added to the compile path when they are available in the target, both PDE+Tycho. So these jar is only meant for compile the `@ProviderType` and alike?

**Reply by tjwatson** [reply]
Yes

**Reply by laeubi** [reply]
Lets give it a try:
- https://github.com/eclipse-equinox/equinox/pull/132
